public/app/features/dashboard-scene/inspect/PanelInspectDrawer.tsx

```diff
@@ -21,39 +22,38 @@ import { InspectTabState } from './types';
 
 interface PanelInspectDrawerState extends SceneObjectState {
   tabs?: Array<SceneObject<InspectTabState>>;
+  panelRef: SceneObjectRef<VizPanel>;
 }
 
 export class PanelInspectDrawer extends SceneObjectBase<PanelInspectDrawerState> {
   static Component = PanelInspectRenderer;
 
-  // Not stored in state as this is just a reference and it never changes
-  private _panel: VizPanel;
+  constructor(state: PanelInspectDrawerState) {
+    super(state);
 
-  constructor(panel: VizPanel) {
-    super({});
-
-    this._panel = panel;
     this.buildTabs();
   }
 
   buildTabs() {
-    const plugin = this._panel.getPlugin();
+    const panel = this.state.panelRef.resolve();
```

Comment:
Not necessarily related to this PR, but I think we can consider a getter instead of `resolve`, similar to how React handles ref. This line would turn into:

```
this.state.dashboardRef.current.setState({ drawer: undefined });
```


or alternatively to avoid confusion with React refs we can do sth like this:

```
this.state.dashboardRef.object.setState({ drawer: undefined });
```

But i think i prefer `current`

Reply:
@dprokop yea, maybe. I tested that but it promotes using it like that in very long hard to read expressions and hides the function call. Having it be a function call encourages a bit more breaking it up. But I agree the consistency with React refs is a good argument